CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml.cs

```diff
@@ -31,12 +34,86 @@ public StartupLanguageSelect()
-            SetAppConfigValue("AppLanguage", LangID);
-            LoadLocale(LangID);
-            NextBtn.IsEnabled = true;
+            tag = tag.ToLower();
+            // English
+            if (tag == "en" || tag.StartsWith("en-")) return "en-us";
+            // Traditional Chinese
+            if (tag == "zh-hant" || tag == "zh-hk" || tag == "zh-mo" || tag == "zh-tw") return "zh-tw";
+            // Chinese defaults to zh-cn
+            if (tag == "zh" || tag.StartsWith("zh-")) return "zh-cn";
+            // German
+            if (tag == "de" || tag.StartsWith("de-")) return "de-de";
+            // Spanish
+            if (tag == "es" || tag.StartsWith("es-")) return "es-419";
+            // Indonesian
+            if (tag == "id" || tag.StartsWith("id-")) return "id-id";
+            // Japanese
+            if (tag == "ja" || tag.StartsWith("ja-")) return "ja-jp";
+            // Korean
+            if (tag == "ko" || tag.StartsWith("ko-")) return "ko-kr";
+            // Polish
+            if (tag == "pl" || tag.StartsWith("pl-")) return "pl-pl";
+            // Portuguese, Portugal
+            if (tag == "pt-pt") return "pt-pt";
+            // Portuguese defaults to pt-br
+            if (tag == "pt" || tag.StartsWith("pt-")) return "pt-br";
+            // Russian
+            if (tag == "ru" || tag.StartsWith("ru-")) return "ru-ru";
+            // Thai
+            if (tag == "th" || tag.StartsWith("th-")) return "th-th";
+            // Vietnamese
+            if (tag == "vi" || tag.StartsWith("vi-")) return "vi-vn";
+            // Default
+            return "en-us";
+        }
+
+        private void RefreshSelection()
+        {
+            SelectLang.SelectedIndex = -1;
+            SelectLang.SelectedIndex = SelectedLangIndex;
+            SelectWindowSize.SelectedIndex = -1;
+            SelectWindowSize.SelectedIndex = SelectedWindowSizeProfile;
+            SelectCDN.UpdateLayout();
+            SelectCDN.SelectedIndex = SelectedCDN;
+        }
+
+        private int SelectedLangIndex
+        {
+            get
+            {
+                var langID = GetAppConfigValue("AppLanguage").ToString() ?? LanguageFallback(CultureInfo.InstalledUICulture.Name);
+                return LanguageNames[langID.ToLower()].LangIndex;
+            }
+            set
+            {
+                if (value < 0) return;
+                var langID = LanguageIDIndex[value].ToLower();
+                if (langID == GetAppConfigValue("AppLanguage").ToString().ToLower()) return;
+                SetAppConfigValue("AppLanguage", langID);
+                LoadLocale(langID);
+
+                // Update the view
+                Bindings.Update();
+                RefreshSelection();
+            }
+        }
+
+        private void SelectLang_OnDropDownOpened(object sender, object e)
+        {
+            // Fix mouse event
+            var incps = InputNonClientPointerSource.GetForWindowId(m_windowID);
```

Comment:
You say "fix mouse event", but what does that actually entail? Is it an issue with the library or C#. Please elaborate in your comment.

Reply:
issue with winui3, the dropdown panel of language selection ComboBox collides with non-client area, making the first item not clickable. Use this to disable whole non-client area temporarily. I can move this to MainWindow.